Doc/library/functions.rst

```diff
@@ -612,9 +618,15 @@ are always available.  They are listed here in alphabetical order.
 
    .. note::
 
-      Most users should just pass a *globals* argument and never *locals*.
-      If exec gets two separate objects as *globals* and *locals*, the code
-      will be executed as if it were embedded in a class definition.
+      When ``exec`` gets two separate objects as *globals* and *locals*, the
+      code will be executed as if it were embedded in a class definition. This
+      means functions defined in the executed code will not be able to access
+      variables assigned at the top level (as the "top level" variables are
+      treated as class variables in a class definition).
+      Passing a :class:`collections.ChainMap` instance as *globals* allows name
+      lookups to be chained across multiple mappings without triggering this
+      behaviour. Values assigned to top-level names in the executed code can be
+      retrieved by passing an empty dictionary as the first entry in the chain.
```

Comment:
This change on `main` triggered a conflict with the PR. While "Most users should just pass a *globals* argument and never *locals*" is technically true, it leaves the question of "What if I need to capture the names bound by the executed code?" question open, so I added a direct reference out to `collections.ChainMap` as a way to collect the writes without changing the code execution semantics.

Reply:
I also added a reference to this PR and comment from https://github.com/python/cpython/pull/119235 (the PR where this note originated)

Reply:
My change here wasn't correct, so I posted a follow-up PR to fix it: https://github.com/python/cpython/pull/119378